include/zenoh-pico/protocol/definitions/network.h

```diff
@@ -62,8 +62,9 @@ typedef _z_qos_t _z_n_qos_t;
 
 static inline _z_qos_t _z_n_qos_create(_Bool express, z_congestion_control_t congestion_control,
                                        z_priority_t priority) {
+    _z_n_qos_t ret;
     _Bool nodrop = congestion_control == Z_CONGESTION_CONTROL_DROP ? 0 : 1;
-    _z_n_qos_t ret = {._val = (uint8_t)((express << 4) | (nodrop << 3) | priority)};
+    ret._val = (uint8_t)((express << 4) | (nodrop << 3) | priority);
```

Comment:
I'd prefer the form 
```
_z_n_qos_t ret = { uint8_t)((express << 4) | (nodrop << 3) | priority };
```
This would guarantee that all fields in `_z_n_qos_t` are set.
Though this variant works too.